cirq-core/cirq/contrib/acquaintance/executor.py

```diff
@@ -100,6 +117,74 @@ def optimization_at(
         )
 
 
+@transformers.transformer
+class StrategyExecutorTransformer:
+    """Executes an acquaintance strategy."""
+
+    def __init__(self, execution_strategy: ExecutionStrategy) -> None:
+        """Initializes transformer.
+
+        Args:
+            execution_strategy: The `ExecutionStrategy` to execute.
+
+        Raises:
+            ValueError: if execution_strategy is None.
+        """
+
+        if execution_strategy is None:
+            raise ValueError('execution_strategy cannot be None')
+        self.execution_strategy = execution_strategy
+        self._mapping = execution_strategy.initial_mapping.copy()
+
+    def __call__(
+        self, circuit: circuits.AbstractCircuit, context: Optional['cirq.TransformerContext'] = None
+    ) -> circuits.Circuit:
+        """Executes an acquaintance strategy using cirq.map_operations_and_unroll and
+        mutates initial mapping.
+
+        Args:
+            circuit: 'cirq.Circuit' input circuit to transform.
+            context: `cirq.TransformerContext` storing common configurable
+              options for transformers.
+
+        Returns:
+            A copy of the modified circuit after executing an acquaintance
+              strategy on all instances of AcquaintanceOpportunityGate
+        """
+
+        circuit = transformers.expand_composite(
+            circuit, no_decomp=expose_acquaintance_gates.no_decomp
+        )
+        expose_acquaintance_gates(circuit)
+        return transformers.map_operations_and_unroll(
+            circuit=circuit,
+            map_func=self._map_func,
+            deep=context.deep if context else False,
+            tags_to_ignore=context.tags_to_ignore if context else (),
+        )
```

Comment:
Add an `unfreeze` to convert the abstract circuit into a circuit and resolve the typing error. 
```suggestion
        ).unfreeze(copy=False)
```